homeassistant/components/adax/data_handler.py

```diff
@@ -0,0 +1,52 @@
+"""Data handler for Adax wifi-enabled home heaters."""
+
+from adax import Adax
+from adax_local import Adax as AdaxLocal
+
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_TOKEN
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_ID, CONNECTION_TYPE, LOCAL
+
+
+class AdaxDataHandler:
+    """Representation of a heater data handler."""
+
+    def __init__(self, entry, hass):
+        """Initialize the data handler."""
+        if entry.data.get(CONNECTION_TYPE) == LOCAL:
+            self._adax = AdaxLocal(
+                entry.data[CONF_IP_ADDRESS],
+                entry.data[CONF_TOKEN],
+                websession=async_get_clientsession(hass, verify_ssl=False)
+            )
+        else:
+            self._adax = Adax(
+                entry.data[ACCOUNT_ID],
+                entry.data[CONF_PASSWORD],
+                websession=async_get_clientsession(hass)
+            )
+
+        self._rooms = None
+
+    async def async_update(self):
+        """Get the latest data."""
+        self._rooms = await self._adax.get_rooms()
```

Comment:
AdaxLocal does not have any get_rooms

Reply:
Will now return None if interface is local